src/lib/semantic-versioning.ts

```diff
@@ -0,0 +1,39 @@
+import { isArray } from "lodash"
+
+export type SemanticVersionNumber = {
+  major: number
+  minor: number
+  patch: number
+}
+
+export function getEigenVersionNumber(
+  userAgent?: string | string[]
+): SemanticVersionNumber | null {
+  if (!userAgent) return null
+  if (!userAgent.includes("Artsy-Mobile")) return null
+
+  const agent = isArray(userAgent) ? userAgent[0] : userAgent
+  const parts = agent.split("/")
+  const version = parts.at(-1)
```

Comment:
**L15** is interesting — [MP seems convinced](https://github.com/artsy/metaphysics/blob/70fd9642465eaaaceff2babb493d0f60d7a7fd84/src/types/graphql.d.ts#L23-L24) that a user agent could be an _array_ of strings as well as just a single string — but I'm not sure why. 

So I just consider the first string in such a case — safe?

**L16-17** is based on my reading of this code in Eigen, and seems to work out fine IRL:

https://github.com/artsy/eigen/blob/396ab8c1cf4a16c8095a3ce240e59a112b496ce0/src/app/store/GlobalStore.tsx#L111-L113

Reply:
I suppose the extra snippet that checks `isArray` is safe to include, but I think that typing is possibly wrong. There's a decent amount of uses of `userAgent` around the codebase (not typed with `ResolverContext`) that treat it as a string and I don't ever recall an issue. It's just a value from a header which should be a string - https://github.com/artsy/metaphysics/blob/f05d2dcf28b915156ae900b185a29830f5b5e378/src/index.ts#L152

Reply:
Cool, casting that to a singular string in 772de17